front/cypress/e2e/admin/users/seat-billing.cy.ts

```diff
@@ -0,0 +1,708 @@
+import { randomString, randomEmail } from '../../../support/commands';
+import { generateProjectFolder } from '../../../fixtures';
+
+describe('Seat based billing', () => {
+  let createdUserIds: string[] = [];
+  let adminAndmoderatorsCount: number;
+
+  type CreateUserType = {
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
+  };
+
+  const createUsers = (users: CreateUserType[]) => {
+    users.forEach((user) => {
+      cy.apiSignup(
+        user.firstName,
+        user.lastName,
+        user.email,
+        user.password
+      ).then((response) => {
+        createdUserIds.push(response.body.data.id);
+      });
+    });
+  };
+
+  const cleanUp = () => {
+    createdUserIds.forEach((id) => {
+      cy.apiRemoveUser(id);
+    });
+    createdUserIds = [];
+  };
+
+  describe('with admin seats', () => {
+    // User 1
+    const user1FirstName = randomString();
+    const user1LastName = randomString();
+    const user1Email = randomEmail();
+    const user1Password = randomString();
+
+    before(() => {
+      createUsers([
+        {
+          firstName: user1FirstName,
+          lastName: user1LastName,
+          email: user1Email,
+          password: user1Password,
+        },
+      ]);
+
+      cy.apiGetUsersCount().then((response) => {
+        adminAndmoderatorsCount =
+          response.body.administrators_count + response.body.managers_count;
+      });
+
+      cy.setAdminLoginCookie();
+    });
+
+    after(() => {
+      cleanUp();
+    });
+
+    it('allows admin to give admin rights or change another an admin to a normal user and shows confirmation when exceeding adminseats', () => {
+      cy.apiGetAppConfiguration().then((appConfigurationResponse) => {
+        let additionalAdmins =
+          appConfigurationResponse.body?.data.attributes.settings.core
+            .additional_admins_number;
+        let maximumAdmins =
+          appConfigurationResponse.body?.data.attributes.settings.core
+            .maximum_admins_number;
+
+        cy.apiGetSeats().then((seatsResponse) => {
+          let usedSeats = seatsResponse.body.data.attributes.admins_number;
+          let totalSeats = additionalAdmins + maximumAdmins;
+          let remainingSeats = totalSeats - usedSeats;
+
+          cy.visit('/admin/users/admins-managers');
+          cy.acceptCookies();
+          cy.get('[data-cy="e2e-admin-and-moderator-count"]').contains(
+            `${adminAndmoderatorsCount}`
+          );
+          cy.get('[data-cy="e2e-admin-remaining-seats"]').contains(
+            `${remainingSeats}`
+          );
+          cy.get('[data-cy="e2e-admin-used-seats"]').contains(`${usedSeats}`);
+          cy.get('[data-cy="e2e-admin-total-seats"]').contains(`${totalSeats}`);
+
+          // Navigate to users page
+          cy.visit('/admin/users');
+
+          // Set user as admin
+          cy.get('.e2e-user-table')
+            .find('.e2e-user-table-row')
+            .first()
```

Comment:
This is repeated 7-8x in this file. Maybe abstract?

Reply:
Sometimes that makes e2e tests harder to follow but I can give it a go